src/axolotl/integrations/nvfp4/kernels/nvfp4_fused_ce.py

```
@@ -0,0 +1,301 @@
+    ``vocab_block`` (env > arg > 4096) is written back to ``_VOCAB_BLOCK`` once at
+    load time; safe since the block width is loss-invariant.
+    """
+    global _VOCAB_BLOCK
+    effective_block = _resolve_vocab_block(vocab_block)
+    _VOCAB_BLOCK = effective_block
+
+    causal = model
+    if hasattr(model, "get_base_model"):
+        try:
+            causal = model.get_base_model()
+        except Exception:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)